src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -364,7 +364,8 @@ private static IUnboundLambdaExpression CreateUnboundLambdaOperation(UnboundLamb
             // an IUnboundLambdaExpression, you need to look at the parent IConversionExpression.
             ITypeSymbol type = null;
             Optional<object> constantValue = ConvertToOptional(unboundLambda.ConstantValue);
-            return new UnboundLambdaExpression(isInvalid, syntax, type, constantValue);
+            Lazy<ILambdaExpression> internalLambda = new Lazy<ILambdaExpression>(() => CreateBoundLambdaOperation(unboundLambda.BindForErrorRecovery()));
```

Comment:
Looks like when the semantic model requests the bound nodes for a given tree, it'll do the error recovery to pretend that there was a boundlambda there all along. https://github.com/dotnet/roslyn/blob/features/ioperation/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs#L159. I'll add an assertion to the SemanticModel of this behavior.